EventFilter/GEMRawToDigi/plugins/GEMRawToDigiModule.cc

```diff
@@ -118,25 +119,69 @@ void GEMRawToDigiModule::produce(edm::StreamID iID, edm::Event& iEvent, edm::Eve
 
     if (nWords < 5)
       continue;
-    const unsigned char* data = fedData.data();
 
-    const uint64_t* word = reinterpret_cast<const uint64_t*>(data);
+    // trailer checks
+    FEDTrailer trailer(fedData.data() + fedData.size() - FEDTrailer::length);
+    if (not trailer.check()) {
+      edm::LogWarning("GEMRawToDigiModule") << "    FED trailer check failed\n";
+    }
+    if (trailer.fragmentLength() * sizeof(uint64_t) != fedData.size()) {
+      edm::LogWarning("GEMRawToDigiModule") << "    FED fragment size mismatch: " << trailer.fragmentLength()
+                                            << " (fragment length) vs " << nWords << " (data size) words\n";
+    }
 
+    const unsigned char* data = fedData.data();
+    const uint64_t* word = reinterpret_cast<const uint64_t*>(data);
     auto amc13Event = gemRawToDigi_->convertWordToAMC13Event(word);
 
-    if (amc13Event == nullptr)
+    if (amc13Event == nullptr) {
+      LogDebug("GEMRawToDigiModule") << "amc13Event FAILED to unpack";
       continue;
+    }
+    if (gemRawToDigi_->amcError() || gemRawToDigi_->vfatError()) {
+      edm::LogWarning("GEMRawToDigiModule")
+          << " amcError:" << gemRawToDigi_->amcError() << " vfatError:" << gemRawToDigi_->vfatError();
+    }
+    // compare trailers
+    if (amc13Event->fragmentLength() != trailer.fragmentLength()) {
+      edm::LogWarning("GEMRawToDigiModule") << " fragmentLength miss match";
```

Comment:
typo: miss match -> mismatch (also in next few lines)

Reply:
in all of these messages (also in the next file), it might be useful to print the actual mismatched values as well

Reply:
thanks @kpedro88! 
I'll fix them after the test

Reply:
The values for amc13Event and trailer would not be that useful, since they should be reading the same data.
This test is to just make sure that they are both reading the last data line.